### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by kpcyrd**
This contains the work of #436 with minor fixes. Moving this into a PR from a local branch causes the github actions to execute even though this isn't merged yet.

Resolves #436

### CODE REVIEW

**Comment by kpp** [comment]
We can use https://github.com/taiki-e/pin-project/pull/117/files#diff-3515bd2b4297af2de6456cf6df7c9f4cR1-R10 to integrate GA with bors.

File: .github/bors.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+status = [
+    "test (ubuntu-latest, stable)",
+    "test (macos-latest, stable)",
+    "test (windows-latest, stable)",
+    "rustfmt",
+    "clippy",
+    "continuous-integration/appveyor/branch"
```
Comment (kpp) [inline .github/bors.toml:7]: We don't need "continuous-integration/appveyor/branch". There are some CI checks in that config we can migrate later.

Comment (kpcyrd) [reply .github/bors.toml]: Done!

**Review by kpp — APPROVED** [review]

**Comment by kpcyrd** [comment]
Appveyor is lagging behind but should finish eventually, I think I'm still not able to merge this until travis is removed though. Preferably only bors would be required.

**Comment by Dylan-DPC** [comment]
I think we might have to nuke t he CI and merge this directly rather than fixing the CI

**Comment by kpp** [comment]
I agree with @Dylan-DPC. The whole point of the original PR is to switch from stuck CI to GA. There are PRs that are waiting for TravisCI status for weeks.

**Comment by kpcyrd** [comment]
I think the problem are the branch protection rules that are defined on github and not part of the repo:

image

I don't have a repo with recent builds to demonstrate this unfortunately, but there should be a list of checks that are mandatory and somebody with admin access can remove travis and appveyor from there.

**Comment by kpp** [comment]
bors: r+

**Comment by dnaq** [comment]
I removed Travis and appveyor from the list.

**Comment by Dylan-DPC** [comment]
well you can remove them, but they are still in bors configuration i guess which makes things a mess.

**Comment by Dylan-DPC** [comment]
I'm going to assume that this is configured properly (there's no way we can verify it without testing) so I will merge it manually and if any issue persists we can fix it after that. Bors won't work for this because it uses the existing configuration.

**Comment by Dylan-DPC** [comment]
judging by [results on another pr](https://github.com/sodiumoxide/sodiumoxide/actions/runs/326021881) I think the config was done properly

**Comment by kpcyrd** [comment]
bors is still stuck on `Expected — Waiting for status to be reported` at #435 and #438, I'm not sure if our bors config is matching the github actions names correctly.

**Comment by kpcyrd** [comment]
Continuing this in #441

**Pull request merged**